staging/vhost-device-sound/src/device.rs

```diff
@@ -351,24 +351,7 @@ impl VhostUserSoundThread {
                         log::error!("{}", Error::from(StreamError::InvalidStreamId(stream_id)));
                         resp.code = VIRTIO_SND_S_BAD_MSG.into();
                     } else {
-                        audio_backend
-                            .write()
-                            .unwrap()
-                            .release(
-                                stream_id,
-                                ControlMessage {
-                                    kind: code,
-                                    code: VIRTIO_SND_S_OK,
-                                    desc_chain,
-                                    descriptor: desc_hdr,
-                                    vring: vring.clone(),
-                                },
-                            )
-                            .unwrap();
-
-                        // PcmRelease needs to flush IO messages; the audio backend will reply when
-                        // it drops the ControlMessage.
-                        continue;
+                        audio_backend.write().unwrap().release(stream_id).unwrap();
```

Comment:
Same as previous commit, match on error to set `resp.code`

Reply:
The only difference here is that I replaced `VIRTIO_SND_S_BAD_MSG` with `VIRTIO_SND_S_IO_ERR` when the state transition is invalid.

Reply:
I think yes :thinking: We need to keep the error semantics, i.e. return IO error when we can't read/write from sound backends.